platform-sdk/swirlds-common/src/test/java/com/swirlds/common/config/sources/MappedConfigSourceTest.java

```diff
@@ -0,0 +1,186 @@
+
+        // when
+        mappedConfigSource.addMapping("b", "not-available");
+
+        // then
+        Assertions.assertEquals(1, mappedConfigSource.getProperties().size());
+        Assertions.assertEquals(Set.of("a"), mappedConfigSource.getPropertyNames());
+    }
+
+    @Test
+    void testAliasesWithSameName() {
+        // given
+        final var configSource = new SimpleConfigSource("a", "1").withValue("b", "2");
+        final var mappedConfigSource = new MappedConfigSource(configSource);
+
+        // when
+        mappedConfigSource.addMapping("c", "a");
+
+        // then
+        Assertions.assertThrows(IllegalArgumentException.class, () -> mappedConfigSource.addMapping("c", "b"));
+    }
+
+    @Test
+    void testWithConfiguration() {
+        // given
+        final var configSource = new SimpleConfigSource("a", "1");
+        final var mappedConfigSource = new MappedConfigSource(configSource);
+
+        mappedConfigSource.addMapping("foo.a", "a");
+
+        final var config =
+                new TestConfigBuilder().withSource(mappedConfigSource).getOrCreateConfig();
+
+        // when
+        final var valueFromOldName = config.getValue("a");
+        final var valueFromNewName = config.getValue("foo.a");
+
+        // then
+        Assertions.assertEquals(valueFromOldName, valueFromNewName);
+    }
+
+    @Test
+    void testNoMappingForEmptySource() {
+        // given
+        final var configSource = new SimpleConfigSource();
+        final var mappedConfigSource = new MappedConfigSource(configSource);
+
+        mappedConfigSource.addMapping("foo.a", "a");
+
+        // when
+        final var config =
+                new TestConfigBuilder().withSource(mappedConfigSource).getOrCreateConfig();
+
+        // then
+        Assertions.assertThrows(NoSuchElementException.class, () -> config.getValue("a"));
+        Assertions.assertThrows(NoSuchElementException.class, () -> config.getValue("foo.a"));
+    }
+
+    @Test
+    void testDefaultValuesForEmptySource() {
+        // given
+        final var configSource = new SimpleConfigSource();
+        final var mappedConfigSource = new MappedConfigSource(configSource);
+        mappedConfigSource.addMapping("foo.a", "a");
+
+        final var config =
+                new TestConfigBuilder().withSource(mappedConfigSource).getOrCreateConfig();
+
+        // when
+        final var oldName = config.getValue("a", "1");
+        final var newName = config.getValue("foo.a", "2");
+
+        // then
+        Assertions.assertEquals("1", oldName);
+        Assertions.assertEquals("2", newName);
+    }
+
+    @Test
+    void testNoMappingIfAlreadyInSource() {
+        // given
+        final var configSource = new SimpleConfigSource("a", "1").withValue("foo.a", "2");
+        final var mappedConfigSource = new MappedConfigSource(configSource);
+
+        // when
+        mappedConfigSource.addMapping("foo.a", "a");
```

Comment:
Is this supposed to create an exception?   It seems really easy to make a mistake and assume a mapping took place.

Reply:
This should not throw an exception. In future we might have support for mutable configuration. In such case a config property could be added at runtime. Next to this the config properties are read when the configuration is created. At the point when a mapping is added the property file has not even be read.